jrpc/internal/service.go

```diff
@@ -187,6 +187,50 @@ func (s *jasperService) Wait(ctx context.Context, id *JasperProcessID) (*Operati
 	}, nil
 }
 
+func (s *jasperService) Restart(ctx context.Context, id *JasperProcessID) (*OperationOutcome, error) {
+	proc, err := s.manager.Get(ctx, id.Value)
+	if err != nil {
+		err = errors.Wrapf(err, "problem finding process '%s'", id.Value)
+		return &OperationOutcome{
+			Success:  false,
+			Text:     err.Error(),
+			ExitCode: -2,
+		}, err
+	}
+
+	// See restartProcess() in rest_service.go.
+	cctx, cancel := context.WithCancel(context.Background())
```

Comment:
So this requires a bit of following of links via comments. This suggests looking in `rest_service.go`'s analogous code:
```go
// Spawn a new context so that the process' context is not potentially
// canceled by the request's. See how createProcess() does this same thing.
ctx, cancel := context.WithCancel(context.Background())
if err := proc.Restart(ctx); err != nil {
    writeError(rw, gimlet.ErrorResponse{
                 StatusCode: http.StatusBadRequest,
                 Message:    err.Error(),
    })
    cancel()
    return
}
```

This makes us look at `createProcess()` but gives the crux of the reasoning here, which is that the context of the request should not be the context that the process itself uses (in particular for this case, for its `reactor()`).

Looking at `createProcess`, which eventually triggers a `reactor()` call itself (in this specific case):
```go
var ctx context.Context
var cancel context.CancelFunc
if opts.Timeout > 0 {
    ctx, cancel = context.WithTimeout(context.Background(), opts.Timeout)
} else {
    ctx, cancel = context.WithCancel(context.Background())
}
```
We see the use of a new root context. This is required because otherwise the new process will rely on the context from the request, which could be canceled or timeout.

Without this, we would expect several tests to fail when the context used in the request is canceled or times out.

Reply:
Sounds good, can you add a comment so it's more clear here?

Reply:
To be sure, are you proposing a comment to be added to `jrpc/internal/service.go`, or an addition of a comment to the logic in `createProcess()`?